### PR TITLE
Fix some wording and asyncio

### PR DESCRIPTION
**Pull request by neubig**
# Description

This fixes wording up to the dataset generator, and moves the `nest_asyncio` out of the jupyter notebook into our library.

Note that I'm reverting the wording around the `.env` file, as the current wording may still encourage people to add the key to their jupyter notebook, which is not the best practice.

# Blocked by

- NA

### CODE REVIEW

File: notebook_demo.ipynb [hunk 1]
```diff
@@ -43,22 +43,12 @@
...
-   "outputs": [],
-   "source": [
-    "!echo \"OPENAI_API_KEY=<your key here>\" >> .env"
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "metadata": {},
-   "source": [
+    "Set your OpenAI API key as an environment variable. A good way to do this is to create a `.env` file with a single line.\n",
```
Comment (zhaochenyang20) [inline notebook_demo.ipynb:46]: I use `echo` in that it's easy to do in Colab. I am still not clear about how to create files in Colab. And `"!echo \"OPENAI_API_KEY=<your key here>\" >> .env"` may be the easiest way.

File: notebook_demo.ipynb [hunk 2]
```diff
@@ -327,62 +346,15 @@
...
-     "output_type": "stream",
-     "text": [
-      "Requirement already satisfied: nest-asyncio in /Users/zhaochen20/miniconda3/envs/prompt/lib/python3.11/site-packages (1.5.6)\n",
-      "Note: you may need to restart the kernel to use updated packages.\n"
-     ]
-    }
-   ],
-   "source": [
-    "%pip install nest-asyncio"
+    "Note that there are a number of hyperparameters here. These are in general good defaults, but you might want to play with them. In particular, this is generating 5,000 examples which may be expensive, so you could choose to generate fewer."
```
Comment (zhaochenyang20) [inline notebook_demo.ipynb:357]:
```suggestion
    "Note that there are a number of hyperparameters here. These are in general good defaults, but you might want to play with them. In particular, this generates 5,000 examples, which may be expensive (roughly $ 5), so you could choose to generate fewer."
```

File: notebook_demo.ipynb [hunk 3]
```diff
@@ -392,22 +364,15 @@
    "outputs": [],
    "source": [
     "from prompt2model.dataset_generator import OpenAIDatasetGenerator, DatasetSplit\n",
-    "from prompt2model.utils.logging_utils import get_formatted_logger\n",
-    "import logging, nest_asyncio\n",
-    "\n",
-    "nest_asyncio.apply()\n",
-    "\n",
-    "generator_logger = get_formatted_logger(\"DatasetGenerator\")\n",
-    "generator_logger.setLevel(logging.INFO)\n",
```
Comment (zhaochenyang20) [inline notebook_demo.ipynb:401]: Why delete the logger? 🤔

File: pyproject.toml [hunk 4]
```diff
@@ -44,6 +44,7 @@ dependencies = [
     "termcolor==2.3.0",
     "psutil==5.9.5",
     "protobuf==3.20.0",
+    "nest-asyncio",
```
Comment (zhaochenyang20) [inline pyproject.toml:47]: We do not need to add a version for this. 🤔

**Review by zhaochenyang20 — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 4 above)
Comment (neubig) [reply pyproject.toml:47]: See https://github.com/neulab/prompt2model/issues/296

File: notebook_demo.ipynb (hunk 3 above)
Comment (neubig) [reply notebook_demo.ipynb:401]: Two reasons:
1. It was very verbose and output many logging lines
2. We want to keep the code in the Jupyter notebook as simple as possible so it doesn't overwhelm readers.

File: notebook_demo.ipynb (hunk 1 above)
Comment (neubig) [reply notebook_demo.ipynb:46]: Yeah, let's revisit this later. There are other options, such as uploading a file: https://stackoverflow.com/a/53296722

**Review by zhaochenyang20 — APPROVED** [review]
LGTM

**Pull request merged**